src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -315,13 +315,9 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
 
                     //  If this is a remove operation, then add any globs that will be removed
```

Comment:
Consider adding a high level comment summarizing what `ComputeItems` does.

Because this is a beefier function, here's an idea. For the summary:
// Compute items does some stuff.
// 1. Does A because B
// 2. Does C because D

Then in the code, when A or C happens you can place another comment:
// 1. Here is some more detail on what A is.

and later when C is happening
//2. Perform C to do a thing.

Reply:
Emphasizing this comment.